jobserv/notify.py

```diff
@@ -132,6 +134,35 @@ def notify_build_complete(build, to_list):
     _send(msg)
 
 
+def notify_build_complete_webhook(build, webhook_url, secret):
+    # try to validate URL certificate before sending POST?
+    # add secret header
+    headers = {
+        'X-JSrv-Signature-256 ': hashlib.sha256(secret).hexdigest()
+    }
+    # build JSON payload
+    payload = {
+        'project_name': build.project.name,
+        'build_id': build.build_id,
+        'build_status': build.status.name,
+        'build_url': build_url(build),
+        'build_reason': build.reason,
+        'runs': []
+    }
+    for run in build.runs:
+        payload['runs'].append({
+            'url':run_url(run),
+            'name': run.name,
+            'status': run.status.name
+            })
+    try:
+        requests.post(webhook_url, data=payload, headers=headers)
+    except requests.exceptions.RequestException:
+        # what to do here?
+        pass
```

Comment:
I'm tempted to say you could fail the build. However, I'm about to recommend doing something I rarely recommend: This webhook call might be slow and its currently being from the context of an API request. We currently don't have any redis/rabbit type queue in this project. How about we create a thread and do the upload from that. You could maybe put some simple retry/sleep type logic and then if it fails, just log an error.